src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java

```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
     public Book borrowBook(Long bookId, Long clientId) {
         Optional<Client> optionalClient = clientRepository.findById(clientId);
         Optional<Book> optionalBook = bookRepository.findById(bookId);
-        if (optionalClient.isPresent() && optionalBook.isPresent()) {
+        //added borrow validation
+        if (optionalClient.isPresent() && optionalBook.isPresent()&&optionalClient.get().getBooks().size()<2 ){
             Client client = optionalClient.get();
             Set<Book> bookSet = client.getBooks();
+            
             bookSet.add(optionalBook.get());
             client.setBooks(bookSet);
             clientRepository.save(client);
             return optionalBook.get();
         }
+        System.out.println("No books for u pilgrim");
```

Comment:
Avoid System.out in production code use Logger

Reply:
expection added